### ISSUE TITLE
FeatureRequest: Add hooks

### ISSUE THREAD

**Issue by hoegaarden** [aptly-dev/aptly#388]
TL;DR:
I'd like to have an option to add hooks when certain things are done, in my special case I'd like to have some hook for an update of a published repo.

I have aplty's API running behind an nginx and use that for pushing some packages from the CI system to aplty and potentially update a local repo. However I'd like to e.g. synchronize that repo to different regions and therefore would like to have aptly call out (to either an API or to a local command).
I don't want to push that to the CI system, as I have multiple ways how packages get pushed to aptly; I want it to be the responsibility of aplty to call out when "something" happened. Basically in the end I want to have a possibility to specify some hook for each(?) action exposed via API. I am happy when this is a single command in the aptly.conf which then needs to do the "routing" and is run as the aptly user.

**Comment by sandrotosi** [comment]
i would love to have this feature too!

we prefer to use the command-line to add packages, publish etc, but we would like to have hooks available for reporting and monitoring (like when someone pushes a package to some repo, etc)

**Comment by smira** [comment]
Thanks, I'll take a look for 1.2.0 release!

**Comment by aigr** [comment]
+1

**Comment by twaik** [comment]
+1
